code/game/turfs/closed/wall/window_frames.dm

```diff
@@ -54,6 +59,102 @@
 
 	return FALSE
 
+///Called by the climbable element if you try climb up. Better hope you're well protected against shocks! XD
+/obj/structure/window_frame/proc/on_try_climb(mob/climber)
+	shock(climber, 100)
+
+///Gives the user a shock if they get unlucky (Based on shock chance)
+/obj/structure/window_frame/proc/shock(mob/user, shock_chance)
+	if(!has_grille) // no grille? dont shock.
+		return FALSE
+	if(!underlaying_cable)
+		return FALSE
+	if(!prob(shock_chance))
+		return FALSE
+	if(!in_range(src, user))//To prevent TK and mech users from getting shocked
+		return FALSE
+	var/turf/my_turf = get_turf(src)
+	var/obj/structure/cable/underlaying_cable = my_turf.get_cable_node()
+	if(electrocute_mob(user, underlaying_cable, src, 1, TRUE))
+		var/datum/effect_system/spark_spread/spark_effect = new /datum/effect_system/spark_spread
+		spark_effect.set_up(3, 1, src)
+		spark_effect.start()
+		return TRUE
+	return FALSE
+
+/obj/structure/window_frame/proc/on_entered(datum/source, AM as mob|obj)
```

Comment:
don't use as, it's a legacy system used for verbs. just type it to atom movable

Reply:
ah, i copy-pasted this proc from somewhere else, I'll change it.